### PR TITLE
101-vm-simple-linux - Update

### PR DESCRIPTION
**Pull request by Krisnatagoras**
### Best Practice Checklist
Check these items before submitting a PR... See the Contribution Guide for the full detail: https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/README.md 

1. uri's compatible with all clouds (Stack, China, Government)
1. Staged artifacts use _artifactsLocation & _artifactsLocationSasToken
1. Use a parameter for resource locations with the defaultValue set to resourceGroup().location
1. Folder names for artifacts (nestedtemplates, scripts, DSC)
1. Use literal values for apiVersion (no variables)
1. Parameter files (GEN-UNIQUE for value generation and no "changemeplease" values)
1. $schema and other uris use https
1. Use uniqueString() whenever possible to generate names for resources.  While this is not required, it's one of the most common failure points in a deployment. 
1. Update the metadata.json with the current date

For details: https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/best-practices.md

- [ ] - Please check this box once you've submitted the PR if you've read through the Contribution Guide and best practices checklist.

### Changelog

*
*
*

### CODE REVIEW

**Comment by bmoore-msft** [comment]
@Krisnatagoras can you check the merge conflicts

**Comment by yane3628** [comment]
This PR reverts the SSH key authentication option. Please keep authentication type with SSH key as the default option.

File: 101-vm-simple-linux/README.md [hunk 1]
```diff
@@ -1,4 +1,4 @@
-# Very simple deployment of a Linux VM
+# Linux Ubuntu 18.04-LTS
```
Comment (bmoore-msft) [inline 101-vm-simple-linux/README.md]: This template will deploy whatever version the user selects, so this needs to be reverted or a more generic option chosen

File: 101-vm-simple-linux/azuredeploy.json [hunk 2]
```diff
@@ -1,247 +1,273 @@
...
+        },
+        "adminPasswordOrKey": {
+            "type": "securestring",
+            "metadata": {
+                "description": "SSH Key or password for the Virtual Machine. SSH key is recommended."
+            }
+        },
+        "dnsLabelPrefix": {
+            "type": "string",
+            "defaultValue": "[toLower(concat('dns-',parameters('vmName')))]",
```
Comment (bmoore-msft) [inline 101-vm-simple-linux/azuredeploy.json]: this must be globally unique, independent of the vm name

File: 101-vm-simple-linux/azuredeploy.json [hunk 3]
```diff
@@ -1,247 +1,273 @@
...
+            "type": "string",
+            "defaultValue": "[resourceGroup().location]",
+            "metadata": {
+                "description": "Location for all resources."
+            }
+        },
+        "VmSize": {
+            "type": "string",
+            "defaultValue": "Standard_B2s",
+            "allowedValues": [
```
Comment (bmoore-msft) [inline 101-vm-simple-linux/azuredeploy.json]: why are we restricting the sizes?

Comment (leestott) [reply 101-vm-simple-linux/azuredeploy.json]: @bmoore-msft we dont want to restrict sizes but present a size which works and can be scaled up if required all the Azure4Student Azure Subscription have had Aseries depreciated the SKU pick list no longer contains these so the B series is now the smallest VM permitted. 

So if a student tries to install any VM templates which are A Series the installation fails.

File: 101-vm-simple-linux/README.md (hunk 1 above)
Comment (Krisnatagoras) [reply 101-vm-simple-linux/README.md]: The original description says that will deploy a Ubuntu Linux VM with the default options being Standard_B2s size VM and the 18.04-LTS Version.

File: 101-vm-simple-linux/azuredeploy.json (hunk 2 above)
Comment (Krisnatagoras) [reply 101-vm-simple-linux/azuredeploy.json]: It's now using a uniqueString to get a globally unique name:

"defaultValue": "[toLower(concat('dns-',uniqueString(parameters('vmName'))))]",

File: 101-vm-simple-linux/README.md (hunk 1 above)
Comment (Krisnatagoras) [reply 101-vm-simple-linux/README.md]: # Very simple deployment of a Linux Ubuntu VM, perhaps?

File: 101-vm-simple-linux/azuredeploy.json [hunk 4]
```diff
@@ -1,247 +1,273 @@
...
-        "name": "[variables('storageAccountType')]"
-      },
-      "kind": "Storage",
-      "properties": {}
+    "$schema": "https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#",
+    "contentVersion": "1.0.0.0",
+    "parameters": {
+        "vmName": {
+            "type": "string",
+            "defaultValue": "[concat('vm', uniqueString(resourceGroup().name))]",
```
Comment (bmoore-msft) [inline 101-vm-simple-linux/azuredeploy.json]: why are we adding uniqueString here?  It won't enable deploying another vm and just make the name more obscure...

File: 101-vm-simple-linux/azuredeploy.json [hunk 5]
```diff
@@ -1,247 +1,273 @@
...
+        },
+        "adminPasswordOrKey": {
+            "type": "securestring",
+            "metadata": {
+                "description": "SSH Key or password for the Virtual Machine. SSH key is recommended."
+            }
+        },
+        "dnsLabelPrefix": {
+            "type": "string",
+            "defaultValue": "[toLower(concat('dns-',uniqueString(parameters('vmName'))))]",
```
Comment (bmoore-msft) [inline 101-vm-simple-linux/azuredeploy.json]: this needs to be globally unique

File: 101-vm-simple-linux/azuredeploy.json (hunk 4 above)
Comment (Krisnatagoras) [reply 101-vm-simple-linux/azuredeploy.json]: changed.

File: 101-vm-simple-linux/azuredeploy.json (hunk 5 above)
Comment (Krisnatagoras) [reply 101-vm-simple-linux/azuredeploy.json]: Changed for:             
"defaultValue": "[toLower(concat('dns-',uniqueString(resourceGroup().id)))]",

File: 101-vm-simple-linux/azuredeploy.json (hunk 3 above)
Comment (bmoore-msft) [reply 101-vm-simple-linux/azuredeploy.json]: We should just note what restrictions are present for the Azure4Student subs, they won't apply to the other subscriptions and restricting the sizes will cause other problems (location, capacity, policy, quota, maintenance) that it's best not to create.  Setting a default is fine, but let's remove allowed values

File: 101-vm-simple-linux/azuredeploy.json [hunk 6]
```diff
@@ -1,247 +1,273 @@
...
+        },
+        "adminPasswordOrKey": {
+            "type": "securestring",
+            "metadata": {
+                "description": "SSH Key or password for the Virtual Machine. SSH key is recommended."
+            }
+        },
+        "dnsLabelPrefix": {
+            "type": "string",
+            "defaultValue": "[toLower(concat('dns-',uniqueString(resourceGroup().id)))]",
```
Comment (bmoore-msft) [inline 101-vm-simple-linux/azuredeploy.json]: use a prefix appropriate for the vm

Comment (Krisnatagoras) [reply 101-vm-simple-linux/azuredeploy.json]: Perhaps that is a better name? 
        "dnsLabelPrefix": {
            "type": "string",
            "defaultValue": "[toLower(concat('dns-', parameters('vmName')))]",

I also changed the gettingstarted telling that it needs to be a unique name.

Comment (Krisnatagoras) [reply 101-vm-simple-linux/azuredeploy.json]: Putting some more thoughts, perhaps this is a better prefix:
"defaultValue": "[toLower(concat('fqdn-', parameters('vmName')))]"

Comment (leestott) [reply 101-vm-simple-linux/azuredeploy.json]: @bmoore-msft  this is another template designed for Azure4Student

File: 101-vm-simple-linux/azuredeploy.json [hunk 7]
```diff
@@ -4,7 +4,6 @@
     "parameters": {
         "vmName": {
             "type": "string",
-            "defaultValue": "simpleLinuxVM",
```
Comment (bmoore-msft) [inline 101-vm-simple-linux/azuredeploy.json]: revert this change

File: 101-vm-simple-linux/azuredeploy.json [hunk 8]
```diff
@@ -34,7 +33,7 @@
         },
         "dnsLabelPrefix": {
             "type": "string",
-            "defaultValue": "[toLower(concat('dns-',uniqueString(resourceGroup().id)))]",
+            "defaultValue": "[toLower(concat('fqdn-', parameters('vmName')))]",
```
Comment (bmoore-msft) [inline 101-vm-simple-linux/azuredeploy.json]: use

concat('simplelinuxvm-', uniqueString(resourceGroup().id))

File: 101-vm-simple-linux/azuredeploy.parameters.json [hunk 9]
```diff
@@ -8,6 +8,9 @@
         "adminPasswordOrKey": {
             "value": "GEN-SSH-PUB-KEY"
         },
+        "vmName": {
```
Comment (bmoore-msft) [inline 101-vm-simple-linux/azuredeploy.parameters.json]: remove this

File: 101-vm-simple-linux/azuredeploy.json (hunk 7 above)
Comment (Krisnatagoras) [reply 101-vm-simple-linux/azuredeploy.json]: Thanks Brian... 

Done!

File: 101-vm-simple-linux/azuredeploy.json (hunk 8 above)
Comment (Krisnatagoras) [reply 101-vm-simple-linux/azuredeploy.json]: Also Done.

File: 101-vm-simple-linux/azuredeploy.parameters.json (hunk 9 above)
Comment (Krisnatagoras) [reply 101-vm-simple-linux/azuredeploy.parameters.json]: Also Done!
